src/crypto/SecretStorage.js

```diff
@@ -132,6 +124,49 @@ export default class SecretStorage extends EventEmitter {
         return keyId;
     }
 
+    /**
+     * Signs a given secret storage key with the cross-signing master key.
+     *
+     * @param {string} [keyId = default key's ID] The ID of the key to sign.
+     *     Defaults to the default key ID if not provided.
+     */
+    async signKey(keyId = this.getDefaultKeyId()) {
+        if (!keyId) {
+            throw new Error("signKey requires a key ID");
+        }
+
+        const keyInfoEvent = this._baseApis.getAccountData(
+            `m.secret_storage.key.${keyId}`,
+        );
+        if (!keyInfoEvent) {
+            throw new Error(`Key ${keyId} does not exist in account data`);
+        }
+        const keyInfo = keyInfoEvent.getContent();
+
+        await this._crossSigningInfo.signObject(keyInfo, 'master');
+        await this._baseApis.setAccountData(
+            `m.secret_storage.key.${keyId}`, keyInfo,
+        );
```

Comment:
Something else is generating the key and saving to account data then this is overwriting it with a signed version? Should we be uploading the key to account data before its signed?

Reply:
ah, that makes sense then.